app/src/main/java/com/example/basicstatecodelab/screens/WellnessViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.basicstatecodelab.screens
+
+import androidx.compose.runtime.toMutableStateList
+import androidx.lifecycle.ViewModel
+import com.example.basicstatecodelab.models.WellnessTask
+
+class WellnessViewModel : ViewModel() {
+    private val _tasks = getWellnessTasks().toMutableStateList()
+    val tasks: List<WellnessTask>
+        get() = _tasks
+
+
+    fun remove(item: WellnessTask) {
+        _tasks.remove(item)
+    }
+
+    fun changeTaskChecked(item: WellnessTask, checked: Boolean) =
```

Comment:
I think it's because it would not cause a re-composition as the list (that holds the state) "would not change" as it doesn't monitor the items that itself contains.

Reply:
I revisited this and as we have `var checked by mutableStateOf(initialChecked)` on the WellnessTask your suggestion would work as well.